src/components/tasks/card/SuggestionBox.tsx

```diff
@@ -0,0 +1,103 @@
+import fetch from '@/helperFunctions/fetch';
+import { userDataType } from '@/interfaces/user.type';
+import { GithubInfo } from '@/interfaces/suggestionBox.type';
+
+type Props = {
+    assigneeName: string;
+    showSuggestion: boolean;
+    onClickName: (userName: string) => void;
+};
+
+const SuggestionBox: FC<Props> = ({
+    assigneeName,
+    showSuggestion = false,
+    onClickName = () => undefined,
+}) => {
+    const [loading, setLoading] = useState<boolean>(false);
+    const [suggestions, setSuggestions] = useState<GithubInfo[]>([]);
+
+    const clickHandler = (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
+        onClickName(e.currentTarget.innerText);
+    };
+
+    const fetchUsers = async () => {
+        setLoading(true);
+        const url = `${process.env.NEXT_PUBLIC_BASE_URL}/users?search=${assigneeName}`;
+        const { requestPromise } = fetch({ url });
+        const users = await requestPromise;
+        const usersData = users.data.users;
+        const suggestedUsers: GithubInfo[] = [];
+        usersData.map((data: userDataType) => {
+            suggestedUsers.push({
+                github_id: data.github_id,
+                profileImageUrl: data.picture
+                    ? data.picture.url
+                        ? data.picture.url
+                        : placeholderImageURL
+                    : placeholderImageURL,
+            });
+        });
+
+        setSuggestions(suggestedUsers);
+        setLoading(false);
+    };
+
+    useEffect(() => {
+        let timerId: NodeJS.Timeout;
+        if (assigneeName) {
+            timerId = setTimeout(() => {
+                fetchUsers();
+            }, 200);
+        }
+
+        return () => {
+            setSuggestions([]);
+
+            clearTimeout(timerId);
+        };
+    }, [assigneeName]);
+
+    let suggestionList;
+
+    if (showSuggestion && assigneeName) {
+        if (suggestions.length) {
+            suggestionList = (
+                <ul className={classes['suggestions']}>
+                    {suggestions.map((suggestion: GithubInfo) => {
+                        return (
+                            <li
+                                key={suggestion.github_id}
+                                onClick={clickHandler}
+                            >
+                                <span>{suggestion.github_id}</span>
+                                <img
+                                    src={suggestion.profileImageUrl}
+                                    width={20}
+                                    height={20}
+                                />
+                            </li>
+                        );
+                    })}
+                </ul>
+            );
+        } else {
+            suggestionList = (
+                <span className={classes['no-suggestions']}>
+                    User not found!
+                </span>
+            );
```

Comment:
why does classname contain classes, please find some fruitful word for it, can confuse other dev's

Reply:
I believe 'classes' is self explaining here as to what it represents.

Reply:
fixed @Pratiyushkumar